content/source/docs/enterprise/run/install-software.html.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: "enterprise2"
+page_title: "Installing Software - Runs - Terraform Enterprise"
+sidebar_current: "docs-enterprise2-run-install"
+---
+
+# Installing Custom Software
+
+In some cases, it may be useful to install certain software on the Terraform worker, 
+such as a configuration management tool or cloud CLI.
+
+Terraform's `local-exec` [provisioner](https://www.terraform.io/docs/provisioners/local-exec.html) can be used in Terraform Enterprise (TFE). This will execute commands on the host machine running Terraform. In general, it should be used in conjunction with the resource(s) the software will be used to interact with, as relying on a placeholder (such as a null resource) may result in inconsistencies as to whether the software is installed.
+
+```hcl
+resource "aws_instance" "example" {
+  ami           = "${var.ami}"
+  instance_type = "t2.micro"
+  provisioner "local-exec" {
+    command = <<EOH
+sudo apt-get update
+sudo apt-get install -y ansible
+EOH
+  }
+}
+```
+
+An [explicit `depends_on`](/intro/getting-started/dependencies.html#implicit-and-explicit-dependencies) may sometimes be needed to place the installation correctly in the provisioning sequence.
+
+Using `local-exec` is only recommended where using a more standard provider-based action is not possible or feasible. See [Terraform Runs: Custom and Community Providers](index.html#custom-and-community-providers) in the documentation for information on how to use custom and community providers in TFE.
```

Comment:
In general it is possible to use a `local-exec` to install software that you're just going to use to provision the created resource, such as installing `chef`. So this note usually doesn't apply to what people are after. It does apply to the situation where someone wants to use a piece of installed software to create something else, and we don't recommend that for things that could be created by TF.

> Using software installed via `local-exec` to create unmanaged infrastructure is not recommended. Instead, if possible, use a provider that can manage the resource using a Terraform configuration. See...